3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -1,55 +1,70 @@
-const buttonGetPokemon = document.createElement('button');
-document.body.appendChild(buttonGetPokemon);
-
-buttonGetPokemon.setAttribute('type', 'submit');
-buttonGetPokemon.textContent = 'Get Pokemon!';
-
-const form = document.createElement('form');
-document.body.appendChild(form);
-
-const selectPokemon = document.createElement('select');
-form.appendChild(selectPokemon);
-
-const pokemonImage = document.createElement('img');
-
 async function fetchData(url) {
   // TODO complete this function
   const response = await fetch(url);
-  const data = await response.json();
-  return data.results;
+  if (response.ok) {
+    const result = await response.json();
+    return result;
+  } else {
+    console.log(response.status);
+    return;
+  }
 }
 
-function fetchAndPopulatePokemons(element) {
+function fetchAndPopulatePokemons(pokemonsArray) {
   // TODO complete this function
-  element.forEach((pokemon) => {
+  let selectPokemon = document.querySelector('select');
+
+  pokemonsArray.results.forEach((pokemon, index) => {
     const option = document.createElement('option');
-    selectPokemon.appendChild(option);
-    option.value = pokemon.name;
     option.textContent = pokemon.name;
+    selectPokemon.appendChild(option);
+    option.value = index;
+  });
+  selectPokemon.addEventListener('change', function (event) {
+    index = selectPokemon[selectPokemon.selectedIndex].innerText;
```

Comment:
The `index` variable is not defined.

```suggestion
    const index = selectPokemon[selectPokemon.selectedIndex].innerText;
```